### DISCUSSION TITLE
Should defaultload option persists between queries?

### DISCUSSION THREAD

**Discussion by tkumpumak** [sqlalchemy/sqlalchemy#10477] [Usage Questions]
First of all this might be user error, so let's check that first before opening issue.
I modified our code such that I wanted certain child object to not load some blobs by default.

I added  _.options(defaultload(model.datas).options(noload(Datas.blobs), noload(Datas.blobs2)),)_
No more blobs, but then unexpected side effect raised that this query produced an id, which was used in the next query to get delete object. Which was subsequently deleted. This delete failed due to foreign key issue.

However if I change to _.options(defaultload(model.datas).options(lazyload(Datas.blobs), lazyload(Datas.blobs2)),)_ then it is again possible to do delete.

Question is that should the defaultload persist like that to the next query?

**Comment by zzzeek** [comment]
defaultload() as a "loader option" is just a placeholder, it does nothing.

as far as if your "noload()" options would "persist", the options would persist on objects that were already loaded but would not persist on objects that were brought into memory from a different query that does not have those options.

generally using noload() is a bad idea, a few people rely upon this ill-conceived option so we dont remove it but it's very difficult to reason about and not generally used.   

if you are trying to prevent large blob columns from being loaded, use map them as [deferred](https://docs.sqlalchemy.org/en/20/orm/queryguide/columns.html#orm-queryguide-deferred-declarative).
